events/api.py

```diff
@@ -1474,6 +1506,24 @@ def bulk_update(self, request, *args, **kwargs):
     def create(self, request, *args, **kwargs):
         return super().create(request, *args, **kwargs)
 
+    def finalize_response(self, request, response, *args, **kwargs):
+        # Switch to normal renderer for docx errors.
+        response = super().finalize_response(request, response, *args, **kwargs)
+        if request.accepted_renderer.format != "docx":
+            return response
+
+        if response.status_code != 200:
+            first_renderer = self.renderer_classes[0]()
+            response.accepted_renderer = first_renderer
+            response.accepted_media_type = first_renderer.media_type
+
+        return response
+
+    def paginate_queryset(self, queryset):
+        if self.request.query_params.get('format') == 'docx':
```

Comment:
As `request.accepted_renderer` is a class, its name might change and then this line would need to change, while leaving it like it is will probably never result in any need to change it so this is probably better. Unless there is some better way to check what renderer that is than `type(self.request.accepted_renderer.).__name__ == 'DOCXRenderer'`.